crates/sui/src/ptb/ptb_parser/build_ptb.rs

```diff
@@ -647,54 +638,45 @@ impl<'a> PTBBuilder<'a> {
             }
             CommandToken::PickGasBudget => {
                 let PTBArg::Identifier(i) = command.args.pop().unwrap() else {
-                    bail!(
-                        "expected identifie at index {} in {}",
-                        self.command_index,
-                        self.current_context_name()
-                    );
+                    error!(self, "expected identifie",);
                 };
                 let picker = match i.to_string().as_str() {
                     "max" => GasPicker::Max,
                     "min" => GasPicker::Min,
                     "sum" => GasPicker::Sum,
-                    x => bail!(
-                        "invalid gas picker: {} at index {} in {}",
-                        x,
-                        self.command_index,
-                        self.current_context_name()
-                    ),
+                    x => error!(self, "invalid gas picker: {}", x,),
                 };
-                self.gas_budget.set_gas_picker(picker)?;
+                self.gas_budget
+                    .set_gas_picker(picker)
+                    .map_err(|e| err!(self, "{e}"))?;
             }
             CommandToken::GasBudget => {
                 let PTBArg::U64(budget) = command.args.pop().unwrap() else {
-                    bail!("expected gas budget");
+                    error!(self, "expected gas budget");
                 };
                 self.gas_budget.add_gas_budget(budget);
             }
             CommandToken::File => {
-                bail!("File commands should be removed at this point");
+                error!(self, "File commands should be removed at this point");
             }
             CommandToken::FileStart => {
                 assert!(command.args.len() == 1);
                 let PTBArg::String(file_name) = command.args.pop().unwrap() else {
-                    bail!("expected file name");
+                    error!(self, "expected file name");
                 };
-                self.file_scope.push((file_name, self.command_index));
-                self.command_index = 0;
+                self.context.push_file_scope(file_name);
             }
             CommandToken::FileEnd => {
                 assert!(command.args.len() == 1);
                 let PTBArg::String(file_name) = command.args.pop().unwrap() else {
```

Comment:
`unwrap` here, we might need to revisit it.

Reply:
This `unwrap` is actually fine because we just checked the length above. Interesting question though of if we should have an assert on the argument len -- I think we should since this is an "internal" command and should therefore always be well-formed, but interested in if your thoughts on it.

Reply:
Yes, I think it's fine to have the `assert` there.